e2e/nomostest/git-server.go

```diff
@@ -119,13 +119,18 @@ func isAvailableDeployment(o client.Object) error {
 func gitServer() []client.Object {
 	// Remember that we've already created the git-server's Namespace since the
 	// SSH key must exist before we apply the Deployment.
-	return []client.Object{
-		gitPodSecurityPolicy(),
```

Comment:
Do we need the PodSecurityPolicy on non-PSP clusters? If so, we need to consider using other alternatives.

Reply:
I don't have all the context here, but I don't think we need these resources for non-PSP clusters. Looking back at the commit where this was introduced it seems this was just trying to make it so the git-server is deployable on PSP-enabled clusters.

The e2e/kind presubmits deploy the git server and shouldn't have PSP enabled. Given that, the presubmit tests are exercising this without PSP.

Reply:
SG